keps/sig-storage/3476-volume-group-snapshot/README.md

```diff
@@ -409,14 +433,46 @@ type VolumeGroupSnapshotContentSpec struct {
 
 // OneOf
 type VolumeGroupSnapshotContentSource struct {
-        // Dynamic provisioning of VolumeGroupSnapshot
-        // A list of PersistentVolume names to be snapshotted together
-        // +optional
-        PersistentVolumeNames []string
+	// VolumeHandles is a list of volume handles on the backend to be snapshotted
+	// together. It is specified for dynamic provisioning of the VolumeGroupSnapshot.
+	// This field is immutable.
+	// +optional
+	VolumeHandles []string
+
+	// GroupSnapshotHandles specifies the CSI "group_snapshot_id" of a pre-existing
+	// group snapshot and a list of CSI "snapshot_id" of pre-existing snapshots
+	// on the underlying storage system for which a Kubernetes object
+	// representation was (or should be) created.
+	// This field is immutable.
+	// +optional
+	GroupSnapshotHandles *GroupSnapshotHandles
+}
 
-        // Pre-provisioned VolumeGroupSnapshot
-        // +optional
-        VolumeGroupSnapshotHandle *string
+type GroupSnapshotHandles struct {
+	// VolumeGroupSnapshotHandle specifies the CSI "group_snapshot_id" of a pre-existing
+	// group snapshot on the underlying storage system for which a Kubernetes object
+	// representation was (or should be) created.
+	// This field is immutable.
+	// Required.
+	VolumeGroupSnapshotHandle string
+
+	// VolumeSnapshotHandles is a list of CSI "snapshot_id" of pre-existing
+	// snapshots on the underlying storage system for which Kubernetes objects
+	// representation were (or should be) created.
+	// This field is immutable.
+	// Required.
+	VolumeSnapshotHandles []string
```

Comment:
In the pre-existing case, do we lose the volume id? Or is it a requirement that the CSI driver needs to return the volume id for pre-existing snapshots?

Reply:
In the pre-existing case, csi-snapshotter needs to call GetVolumeGroupSnapshot CSI RPC which requires a group_snapshot_id and a list of individual snapshot_ids. The reason is that some backends may need both. We had lots of back and forth discussions regarding this in the CSI spec review. So the get call is like a validation that just verifies the handles really exist in the backend.